src/main/scala/viper/gobra/reporting/DefaultErrorBackTranslator.scala

```diff
@@ -112,9 +125,23 @@ class DefaultErrorBackTranslator(
     case vprerr.WhileFailed(Source(info), reason, _) =>
       ForLoopError(info) dueTo translate(reason)
     case vprerr.AssertFailed(Source(info), reason, _) =>
-      AssertError(info) dueTo translate(reason)
+      info.origin match {
+        case origin: AnnotatedOrigin =>
+          origin.annotation match {
+            case OverflowCheckAnnotation => OverflowError(info) dueTo translate(reason)
+            case _ => ???
+          }
+        case _ =>
+          AssertError(info) dueTo translate(reason)
+      }
```

Comment:
The split into the default translater and a default translator for annotated code has not happened yet, right? Or am I missing something?

Reply:
I added it to the `defaultReasonTransformer` function but I overlooked the case split in `defaultErrorTransformer` function. Changed that in last commit